cmd/create.go

```diff
@@ -429,6 +434,47 @@ func (cc *createCmd) createFiles(detectedLang *config.DraftConfig, lowerLang str
 	return nil
 }
 
+func (cc *createCmd) detectDefaults(detectedLang *config.DraftConfig, lowerLang string) {
+	if lowerLang == "gradlew" || lowerLang == "gradle" {
+		// read from build.gradle and detect version
+		f, err := os.Open("build.gradle")
+		if err != nil {
+			log.Warn("Unable to read build.gradle, skipping detection")
+			return
+		}
+		defer f.Close()
+		detectedDefaults := make([]config.BuilderVarDefault, 0)
+		scanner := bufio.NewScanner(f)
+		separator := func(c rune) bool { return c == ' ' || c == '=' }
+		cutset := func(c rune) bool { return c == '\'' || c == '"' }
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.Contains(line, "sourceCompatibility") {
+				detectedVersion := strings.FieldsFunc(line, separator)[1] // sourceCompatibility = '1.8'
```

Comment:
💡 Suggestions/Observation: wdyt? Can we please add sample out come and what these 3 lines are trying to do and it will be for understanding for future maintainer and the repeat usage in like `464-467` as well, and also why `[1]` element context as well.

Reply:
Yes I have added comments , please let me know if it looks good. Thank you.